conformance/lang/expressions/string-template-expr/string_template_expr.balt

```diff
@@ -0,0 +1,323 @@
+Labels: string, function-defn, string-template-expr
+
+type T1 "c";
+type T2 "m"|"n";
+type T3 2;
+type T4 2.0|3|5;
+type T5 true;
+
+function init() {
+    T1 a1 = "c";
+    string temp1 = string `${a1}`;
+    io:println(temp1); // @output c
+    T2 a2 = "m";
+    string temp2 = string `${a2}`;
+    io:println(temp2); // @output m
+    T3 a3 = 2;
+    string temp3 = string `${a3}`;
+    io:println(temp3); // @output 2
+    T4 a4 = 5;
+    string temp4 = string `${a4}`;
+    io:println(temp4); // @output 5
+    T5 a5 = true;
+    string temp5 = string `${a5}`;
+    io:println(temp5); // @output true
+}
+
+Test-Case: output
+Description: Test type of the expression in each interpolation of string-template-expr.
+Labels: check, string, function-defn, string-template-expr, int, byte, union-type
+
+function init() {
+    byte a1 = 4;
+    string temp1 = string `${a1}`;
+    io:println(temp1); // @output 4
+    int:Unsigned8 a2 = 12;
+    string temp2 = string `${a2}`;
+    io:println(temp2); // @output 12
+    int:Signed8 a3 = 12;
+    string temp3 = string `${a3}`;
+    io:println(temp3); // @output 12
+    int:Signed16 a4 = 23;
+    string temp4 = string `${a4}`;
+    io:println(temp4); // @output 23
+    int:Unsigned16 a5 = 43;
+    string temp5 = string `${a5}`;
+    io:println(temp5); // @output 43
+    int:Signed32 a6 = 0x33;
+    string temp6 = string `${a6}`;
+    io:println(temp6); // @output 51
+    int:Unsigned32 a7 = 32;
+    string temp7 = string `${a7}`;
+    io:println(temp7); // @output 32
+    byte|int:Unsigned8|int:Signed8|int:Unsigned16|int:Signed16|int:Unsigned32 a8 = 23;
+    string temp8 = string `${a8}`;
+    io:println(temp8); // @output 23
+}
+
+Test-Case: error
+Description: Test type of the expression in each interpolation of string-template-expr.
+Labels: string, string-template-expr, int, record-type, optional-type, tuple-type, function-type
+
+function errorFunction() {
+    record {} a1 = {};
+    string _ = string `${a1}`; // @error expected '(int|float|decimal|string|boolean)', found 'record {| anydata...; |}'
+    function  a2 = f1;
+    string _ = string `${a2}`; // @error expected '(int|float|decimal|string|boolean)', found 'function'
+    () a3 = ();
+    string _ = string `${a3}`; // @error expected '(int|float|decimal|string|boolean)', found '()'
+    string? a4 = "a4";
+    string _ = string `${a4}`; // @error expected '(int|float|decimal|string|boolean)', found 'string?'
+    json a5 = 12;
+    string _ = string `${a5}`; // @error expected '(int|float|decimal|string|boolean)', found 'json'
+    [int, int] a6 = [10, 10];
+    string _ = string `${a6}`; // @error expected '(int|float|decimal|string|boolean)', found '[int,int]'
+}
+
+function f1() returns int {
+    return 10;
+}
+
+Test-Case: error
+Description: Test type of the expression in each interpolation of string-template-expr.
+Labels: string, function-defn, string-template-expr, union-type
```

Comment:
Don't we need `boolean` label

Reply:
No. right? we didn't use the `boolean` type